torch/autograd/function.py

```diff
@@ -35,6 +65,27 @@ def mark_dirty(self, *args):
         should be given to this function, to ensure correctness of our checks.
         It doesn't matter whether the function is called before or after
         modification.
+
+        Examples::
+            >>> class Inplace(Function):
+            >>>     @staticmethod
+            >>>     def forward(ctx, x):
+            >>>         x.add_(1)
+            >>>         ctx.mark_dirty(x)
+            >>>         return x
+            >>>
+            >>>     @staticmethod
+            >>>     @once_differentiable  # grad_input is not a function of input
+            >>>     def backward(ctx, grad_output):
+            >>>         return grad_output
+            >>>
+            >>> a = torch.tensor(1., requires_grad=True, dtype=torch.double).clone()
+            >>> b = a * a
+            >>> Inplace.apply(a)  # This would lead to wrong gradients!
+            >>>                   # but the engine would not know unless we mark_dirty
```

Comment:
Well in this case it would. Because you're using Tensors haha

Maybe in your forward you can go to numpy, inplace modify the numpy array and return the Tensor that shares memory with it?

Reply:
`handle_view_on_rebase` isn't called unless grad mode is enabled no?
Since the forward is done in no-grad mode, the engine does 'know', but just doesn't care?

Reply:
What I mean is that the mark_dirty() Is not strictly needed as long as you only modify inplace Tensors.
If you check the versions in that custom function, you will see that it is bumped twice :D

Reply:
Whoops, good point